packages/edit-widgets/src/components/edit-widgets-initializer/index.js

```diff
@@ -3,7 +3,26 @@
  */
 import Layout from '../layout';
 
+/**
+ * WordPress dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+
 function EditWidgetsInitializer( { settings } ) {
+	// Since this module relies on the idea of using a "stub post", a state must be initialized
+	// first, or the data layer would attempt to retrieve the stub post from the API.
+	const { initializeState } = useDispatch( 'core/edit-widgets' );
+	const [ stateInitialized, setStateInitialized ] = useState( false );
+	useEffect( () => {
+		initializeState();
```

Comment:
It's strange to see an action that looks so... imperative. I wonder if instead of an action we should use a resolver `getStubPost` and then discard the return value? Third party plugins might find the return value useful.

_Not a strong opinion, just an idea, take it with a grain of salt, etc._

Reply:
Oh. There's already `getWidgetAreas` which is called here:

https://github.com/WordPress/gutenberg/blob/2acb853a920e84662e2ea7a3b55ab0f595d5b578/packages/edit-widgets/src/components/widget-areas-block-editor-provider/index.js#L39

If we have that, why is this PR necessary?

Reply:
@noisysocks we just need this PR to explicitly do things in certain order and avoid race conditions. Without this PR the post is requested before the widgets resolved initialized it. We could maybe line up the code so that it doesn’t happen, but I’d rather do it explicitly.